### PR TITLE
feat: Additional parameters in extension functions

### LINKED ISSUE #23: Implement supporting of additional parameters in extension functions
Now we can create mapstruct method only with one parameter, that will become receiver in extension function. I think it would be great, if we could create methods with additional parameters. First parameter as receiver of extension function and others as input parameters.

Example:
```kotlin
data class EmployeeModel(val firstName: String)
data class EmployeeCompanyView(val firstName: String, val company: String)

@Mappings(
  Mapping(target = "company", source = "company")
)
abstract fun toEmployeeCompanyView(employeeModel: EmployeeModel, company: String): EmployeeCompanyView
```
Compiled to:
```kotlin
fun EmployeeModel.toEmployeeCompanyView(company: String) =
    Mappers.getMapper(EmployeeMapper::class.java).toEmployeeCompanyView(this, company)
```
So we can call:
```kotlin
val model = EmployeeModel("Alex")
val view = model.toEmployeeCompanyView("amazon")
```

### PR DESCRIPTION
**Pull request by olxmute**
Closes https://github.com/driver733/mapstruct-fluent/issues/23

### CODE REVIEW

File: examples/example-spring/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt [hunk 1]
```diff
@@ -15,11 +13,19 @@ class EmployeeMapperTest(
...
+        val actualEmployeeView = model.toEmployeeView()
+
+        actualEmployeeView shouldBe EmployeeView("Alex")
+        actualEmployeeView shouldBe mapper.toEmployeeView(model)
+    }
+
+    test("mapstruct additional arguments") {
+        val model = EmployeeModel("Alex")
+
+        val actualView = model.toEmployeeCompanyView("amazon")
```
Comment (driver733) [inline examples/example-spring/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt]: Rename to "actual" (I think that mentioning additional info in such small context is unnecessary)

File: examples/example-spring/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt [hunk 2]
```diff
@@ -15,11 +13,19 @@ class EmployeeMapperTest(
...
+
+        actualView shouldBe EmployeeCompanyView("Alex", "amazon")
+        actualView shouldBe mapper.toEmployeeCompanyView(model, "amazon")
```
Comment (driver733) [inline examples/example-spring/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt]:
```
actual.apply {
  shouldBe ...
  shouldBe ...
}
```

File: processors/processor/src/main/kotlin/com/driver733/mapstructfluent/MapstructFluentExtensionsAnnotationProcessor.kt [hunk 3]
```diff
@@ -31,6 +31,9 @@ class MapstructFluentExtensionsAnnotationProcessor : AbstractProcessor(), Mapper
         fileSpecBuilder(processingEnv, mapper, "FluentExtensions")
 
     override fun process(fileSpecBuilder: FileSpec.Builder, method: ExecutableElement, mapper: Element, src: String?) {
+        val receiver = method.parameters.first().kotlinType()
+        val otherParameters = method.parameters.drop(1).map(VariableElement::toParameterSpec)
```
Comment (driver733) [inline processors/processor/src/main/kotlin/com/driver733/mapstructfluent/MapstructFluentExtensionsAnnotationProcessor.kt]: Common logic should be moved to the GenericProcessor class

File: processors/processor-spring/src/main/kotlin/com/driver733/mapstructfluent/MapstructSpringFluentExtensionsAnnotationProcessor.kt [hunk 4]
```diff
@@ -29,7 +30,10 @@ class MapstructSpringFluentExtensionsAnnotationProcessor : AbstractProcessor(),
     override fun fileSpecBuilder(mapper: Element) =
         fileSpecBuilder(processingEnv, mapper, "FluentSpringExtensions")
 
-    override fun process(fileSpecBuilder: FileSpec.Builder, method: ExecutableElement, mapper: Element, src: String?) =
+    override fun process(fileSpecBuilder: FileSpec.Builder, method: ExecutableElement, mapper: Element, src: String?) {
+        val receiver = method.parameters.first().kotlinType()
+        val otherParameters = method.parameters.drop(1).map(VariableElement::toParameterSpec)
```
Comment (driver733) [inline processors/processor-spring/src/main/kotlin/com/driver733/mapstructfluent/MapstructSpringFluentExtensionsAnnotationProcessor.kt]: Common logic should be moved to the GenericProcessor class

File: processors/common/src/main/kotlin/com/driver733/mapstructfluent/TypeUtils.kt [hunk 5]
```diff
@@ -57,3 +57,8 @@ fun TypeName.toKotlinType(): TypeName =
             }
         }
     }
+
+fun Element.toParameterSpec() =
+    ParameterSpec(simpleName.toString(), kotlinType(), emptyList())
+
+fun List<ParameterSpec>.joinToString() = joinToString(prefix = ", ", transform = ParameterSpec::name)
```
Comment (driver733) [inline processors/common/src/main/kotlin/com/driver733/mapstructfluent/TypeUtils.kt]: Function statement should start on the next line (like in the other function statements)

File: examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt [hunk 6]
```diff
@@ -11,13 +11,19 @@ class EmployeeMapperTest : FunSpec({
...
+        actualEmployeeView shouldBe Mappers.getMapper(EmployeeMapper::class.java).toEmployeeView(model)
+    }
+
+    test("mapstruct additional arguments") {
+        val model = EmployeeModel("Alex")
+
+        val actualView = model.toEmployeeCompanyView("amazon")
+
+        actualView shouldBe EmployeeCompanyView("Alex", "amazon")
+        actualView shouldBe Mappers.getMapper(EmployeeMapper::class.java).toEmployeeCompanyView(model, "amazon")
```
Comment (driver733) [inline examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt]:
```
actual.apply {
  shouldBe ...
  shouldBe ...
}
```

File: examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt [hunk 7]
```diff
@@ -11,13 +11,19 @@ class EmployeeMapperTest : FunSpec({
...
+        val actualEmployeeView = model.toEmployeeView()
+
+        actualEmployeeView shouldBe EmployeeView("Alex")
+        actualEmployeeView shouldBe Mappers.getMapper(EmployeeMapper::class.java).toEmployeeView(model)
+    }
+
+    test("mapstruct additional arguments") {
+        val model = EmployeeModel("Alex")
+
+        val actualView = model.toEmployeeCompanyView("amazon")
```
Comment (driver733) [inline examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt]: Rename to actual

File: examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt [hunk 8]
```diff
@@ -11,13 +11,19 @@ class EmployeeMapperTest : FunSpec({
...
         val model = EmployeeModel("Alex")
 
-        model
-            .toEmployeeView()
-            .shouldBe(
-                Mappers
-                    .getMapper(EmployeeMapper::class.java)
-                    .toEmployeeView(model)
-            )
+        val actualEmployeeView = model.toEmployeeView()
```
Comment (driver733) [inline examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt]: rename to actual

File: examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt [hunk 9]
```diff
@@ -11,13 +11,19 @@ class EmployeeMapperTest : FunSpec({
...
+
+        actualEmployeeView shouldBe EmployeeView("Alex")
+        actualEmployeeView shouldBe Mappers.getMapper(EmployeeMapper::class.java).toEmployeeView(model)
```
Comment (driver733) [inline examples/example/src/test/kotlin/com/driver733/mapstructfluent/EmployeeMapperTest.kt]:
```
actual.apply {
  shouldBe ...
  shouldBe ...
}
```

**Review by driver733 — CHANGES_REQUESTED** [review]

File: processors/processor/src/main/kotlin/com/driver733/mapstructfluent/MapstructFluentExtensionsAnnotationProcessor.kt (hunk 3 above)
Comment (olxmute) [reply processors/processor/src/main/kotlin/com/driver733/mapstructfluent/MapstructFluentExtensionsAnnotationProcessor.kt]: I've moved a little bit more code to GenericProcessor, please verify are you okay with that

**Review by driver733 — DISMISSED** [review]
Insightful refactoring, thanks!

**Review by driver733 — APPROVED** [review]

**Pull request merged**